sdk/keyvault/keyvault-keys/review/keyvault-keys.api.md

```diff
@@ -286,6 +293,29 @@ export interface KeyReleasePolicy {
     data?: Uint8Array;
 }
 
+// @public
+export interface KeyRotationLifetimeAction {
+    action: KeyRotationPolicyAction;
+    timeAfterCreate?: string;
```

Comment:
I see these names match the swagger, which is probably fine. There's not quite the analog to our certificate's `LifetimeActions` that use a date; however, I would suggest a swagger change that these use the ISO 8601 *duration* format, which is very different than ISO 8601 date formats used elsewhere in this and many other SDKs.

That said, I think for idiomatic reasons we should consider various languages' / frameworks' `TimeSpan` (.NET) structs; though, I believe @JeffreyRichter had some problem with this as far as Go goes. At the very least, we'll want to document our APIs that this is the ISO 8601 duration format e.g., "P90D" for 90 days, and probably even link to it.

Reply:
Swagger should not have any 8601 string durations in it. We want number in units (seconds, minutes, hours). How this number is represented in various programming languages is fine; .NET can use a Timespan initialized from the number in the proper units. FYI: Go has a time.Duration type too but it is just a thin wrapper over an integer.

Reply:
To Jeffrey's point (sorry - I was misremembering his concern was about the swagger, not the SDK), certificates already has a precedent of requiring a number of days: [`days_before_expiry`](https://github.com/Azure/azure-rest-api-specs/blob/a331a455fd31e022f8df07105058ae586a02ab03/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/certificates.json#L1648) (though, should be camelCase - this was before my time).

Reply:
Thank you for the feedback! @jlichwa - certificates swagger already [uses numbers](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/certificates.json#L1639-L1655) to represent triggers for certificate policy - is this something we can model similarly for key rotation policies?

Right now the keys swagger defines [timeAfterCreate and timeBeforeExpiry](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json#L2231-L2243) as strings representing durations - is that something that can be changed to number in units like we do in certificates?

Thanks in advance

Reply:
/cc @herveyw-msft

Reply:
We purposely deviated from Certificates which were defined long time ago and we did want to repeat same mistakes. Keys model is more flexible with ability to easily define time span in any units (initially we just support years, months, days but later we may extend it to hours, minutes....
Engineers found that ISO 8601 is a standard for TimeSpan - we same in Service BUS https://docs.microsoft.com/en-us/rest/api/servicebus/preview/queues/get#sbqueue

Reply:
It does look like we have some precedence for using durations. At least in JS we have `autoDeleteOnIdle` in service bus and monitor query's `timespan` parameter; however, we should address @JeffreyRichter's guidance. So I propose:

1. Merging this PR once we have support for `rotate` permission in arm template (so that we can run live tests against the service)
2. Setting aside some time next week (week of 8/23) when folks are back in the office to discuss the guidelines here - I'd love to get to resolution specifically for Key Rotation and I'll make issues based on the outcome. I can schedule something for next week

cc @AlexGhiondea